internal/provider/user_role_grants_resource_test.go

```diff
@@ -140,10 +162,11 @@ func testRoleResource(t *testing.T, userId string, useMock bool) {
 		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: getTestRoleFullyPermissionsedResourceConfig(userId),
+				Config: getTestRoleFullyPermissionsedResourceConfig(),
 				Check: resource.ComposeTestCheckFunc(
-					testRoleMembership(resourceNameTest, userId, "CLUSTER_ADMIN", true),
-					resource.TestCheckResourceAttr(resourceNameTest, "id", userId),
```

Comment:
I removed this check because we don't have access to the service account's ID at this point. I added line 197 to make up for this. Not sure if that suffices

Reply:
I don't think its really checking the same thing but passing in the userId as done before also seems unnecessary.  The check you added seems fine to keep.